common/nodejs-utils/src/bin/update_node_inventory.rs

```diff
@@ -0,0 +1,168 @@
+// Required due to: https://github.com/rust-lang/rust/issues/95513
+#![allow(unused_crate_dependencies)]
+
+use anyhow::{Context, Result};
+use heroku_inventory_utils::{
+    checksum::Checksum,
+    inv::{read_inventory_file, Arch, Artifact, Inventory, Os},
+};
+use node_semver::Version;
+use serde::Deserialize;
+use sha2::Sha256;
+use std::{
+    collections::{HashMap, HashSet},
+    env, fs, process,
+};
+
+/// Updates the local node.js inventory.toml with versions published on nodejs.org.
+fn main() {
+    let inventory_path = env::args().nth(1).unwrap_or_else(|| {
+        eprintln!("Usage: update_inventory <path/to/inventory.toml>");
+        process::exit(2);
+    });
+
+    let inventory_artifacts: HashSet<Artifact<Version, Sha256>> =
+        read_inventory_file(&inventory_path)
+            .unwrap_or_else(|e| {
+                eprintln!("Error reading inventory at '{inventory_path}': {e}");
+                std::process::exit(1);
+            })
+            .artifacts
+            .into_iter()
+            .collect();
+
+    let upstream_artifacts = list_upstream_artifacts().unwrap_or_else(|e| {
+        eprintln!("Failed to fetch upstream node.js versions: {e}");
+        process::exit(4);
+    });
+
+    let inventory = Inventory {
+        artifacts: upstream_artifacts,
+    };
+
+    let toml = toml::to_string(&inventory).unwrap_or_else(|e| {
+        eprintln!("Error serializing inventory as toml: {e}");
+        process::exit(6);
+    });
+
+    fs::write(inventory_path, toml).unwrap_or_else(|e| {
+        eprintln!("Error writing inventory file: {e}");
+        process::exit(7);
+    });
+
+    let remote_artifacts: HashSet<Artifact<Version, Sha256>> =
+        inventory.artifacts.into_iter().collect();
+
+    [
+        ("Added", &remote_artifacts - &inventory_artifacts),
+        ("Removed", &inventory_artifacts - &remote_artifacts),
+    ]
+    .iter()
+    .filter(|(_, artifact_diff)| !artifact_diff.is_empty())
+    .for_each(|(action, artifacts)| {
+        let mut list: Vec<&Artifact<Version, Sha256>> = artifacts.iter().collect();
+        list.sort_by_key(|a| &a.version);
+        println!(
+            "{} {}.",
+            action,
+            list.iter()
+                .map(ToString::to_string)
+                .collect::<Vec<_>>()
+                .join(", ")
+        );
+    });
+}
+
+fn list_upstream_artifacts() -> Result<Vec<Artifact<Version, Sha256>>, anyhow::Error> {
+    let target_version = Version::parse("0.8.6").context("Failed to parse version")?;
```

Comment:
Minor - `target_version` is a bit of a misnomer.  

```suggestion
let earliest_version = Version::parse("0.8.6").context("Failed to parse earliest Node.js version")?;
```

Reply:
Ah that was a mistake -- fixed in https://github.com/heroku/buildpacks-nodejs/pull/814/commits/3f1d6cfc51819af8f428f6bfbedbdd3fddd888be :)